### ISSUE TITLE
[static-property-placement] Violation on accessing static variable

### ISSUE THREAD

**Issue by jp7837** [yannickcr/eslint-plugin-react#2283]
Example:
```typescript
class MyComponent extends React.Component<MyProps, MyState> {
	public static displayName = "MyComponent ";
	public myMethod() {
		console.log(MyComponent.displayName);
	}
}
```

In the example, line 4 is flagged as a static-property-placement violation even though it simply accesses the property.

**Comment by ljharb** [comment]
cc @dmason30

**Comment by ljharb** [comment]
With what rule config?

**Comment by dmason30** [comment]
@ljharb Thanks for the cc. I'll get round to this shortly. Clearly the first change is that the rule needs to check that the `MemberExpression` is actually a definition and followed by a value `= 'value'`.

I have never needed to access any of these properties before should we expect that someone is changing these properties in a class? For example:

```
class MyComponent extends React.Component<MyProps, MyState> {
	public static displayName = "MyComponent ";
	public myMethod(new_display_name) {
		MyComponent.displayName = new_display_name;
	}
}
```

Should I ignore `MemberExpression` definitions if they are inside a class/function?

**Comment by ljharb** [comment]
Yes; static members inside methods can be ignored, i think.

**Comment by jp7837** [comment]
Buddy test pass on my local code :)
